source/manual/test-and-build-a-project-with-github-actions.html.md

```diff
@@ -27,22 +27,64 @@ You should name the file used for configuring the CI workflow `ci.yml`.
 
 ### When the CI workflow should run
 
-The workflow should be configured to run on [`push`][push-event] and
-[`pull_request`][pull-request-event] events. The `push` event occurs for every
-push to the repository and this will generate a build against the branch
-that was pushed. The `pull_request` event occurs whenever a pull request
-is opened and it is triggered again any time the pull request changes (such
-as extra commits or rebasing), these differ to the `push` event by merging in
-the main branch before running operations. Pull requests from forks of a
-repo will only trigger the `pull_request` event.
+The workflow should be configured to run when there is a [`push`][push-event]
+to the default branch (typically "main") and when a
+[`pull_request`][pull-request-event] opened. This will mean that there is a
+CI run when a branch is merged (the `push` to main) and it will run against
+any changes introduced in a pull request. We prefer to run against a
+`pull_request` event to all `push` events, as this allows external contributors
+to have pull requests tested and because `pull_request` merges the changes with
+the repository default branch.
 
-For example:
+Example configuration:
 
 ```yml
 # ./github/workflows/ci.yml
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
 ```
 
+To have CI runs be able to run outside of a pull request, for example before a
+pull request is opened, you may configure a repository to have a
+`workflow_dispatch` event so it can be ran from the GitHub Actions user
+interface. If you do this you will need to configure the checkout action to
+reference the appropriate commit.
+
+Example configuration:
+
+```yml
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'The branch, tag or SHA to checkout'
+        required: false
+        type: string
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref || github.ref }}
+```
+
+We previously recommended CI jobs to run on `[push, pull_request]` as per
+[RFC-123][] however this caused an appearance of duplicate runs of a CI job
+of a pull request (one for the `push` another for the `pull_request`)
+which were both superfluous for our testing and a point of confusion.
```

Comment:
```suggestion
which were both superfluous for our testing and a point of confusion, as well as wasting resources and in some cases exacerbating quota issues caused by expensive tests.
```

Reply:
I've put a more succinct, and slightly softer, phrasing in.